src/plugins/mavlink_ftp/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+include_directories(
+    SYSTEM ${PROJECT_SOURCE_DIR}/../third_party/filesystem/include
+)
+
+add_library(mavsdk_mavlink_ftp
+    mavlink_ftp.cpp
+    mavlink_ftp_impl.cpp
+    crc32.cpp
+)
+
+target_compile_options(mavsdk_mavlink_ftp
+  PRIVATE -fexceptions
```

Comment:
:astonished: do we need this? We generally don't use exceptions.

Reply:
I see that you use `try` `catch` around `std::atoi` but that's in the example and not in this plugin.

Reply:
ghc-filesystem library uses exceptions. Maybe it would be better to go back to old ANSI C std functions. std::filesystem doesn't work on all platforms this is why I had to switch to ghc-filesystem. There is too much trouble with it.

Reply:
I am probably wrong, but... greping for `catch(`, it seems to me like the only place using an exception in that library is [here](https://github.com/gulrak/filesystem/blob/master/include/ghc/filesystem.hpp#L4734-L4740). And that's in a `GHC_OS_WINDOWS` `#ifdef`. Could that be?

Reply:
What about using boost filesystem, then? It can apparently be built with `BOOST_NO_EXCEPTION`. Wouldn't it be easier than doing it manually?

Reply:
I guess doing it in C is faster than adding dependencies.

Reply:
I removed the ghc_filesystem dependency.